libraries/nomad_job.rb

```diff
@@ -20,16 +21,40 @@ def initialize(url, job_name)
   end
 
   def running?
-    job = http_json("#{@url}/v1/job/#{@job_name}")
+    job = get_job
     job['Status'] == 'running'
   end
 
+  def job_status
+    job = get_job
+    print_allocation_logs(job['ID'])
+    job_status = job['Status']
+  end
+
   def to_s
     "Nomad job #{@job_name} at #{@url}"
   end
 
   private
 
+  def get_job
+    job = http_json("#{@url}/v1/job/#{@job_name}")
+  end
+
+  def print_allocation_logs(job_id)
+    allocations = http_json("#{@url}/v1/job/#{job_id}/allocations")
+    allocations.each do |allocation|
+        task_states = allocation['TaskStates']
+        task_states.each do |task_state|
```

Comment:
You can rewrite this a bit more clearly like this:

```ruby
task_states.each do |task_name, task_state| # key, value
```
Then, you can use `task_name` instead of `task_state[0]`, and `task_state` instead of `task_state[1]`, which I think is a bit clearer.